### ISSUE TITLE
Create a reference / cheat sheet mapping engine-specific parameters to parsnip ones?

### ISSUE THREAD

**Issue by khughitt** [tidymodels/parsnip#211]
As I'm going through converting algorithm-specific logic to generalized parsnip model specifications, one thing I am thinking that could be useful would be to create one or more reference tables or "cheat sheets" in the parsnip docs describing how to map from implemention-specific parameters (e.g. "eta" in xgboost) to parsnip model parameters (e.g. "learn_rate" in parsnip::boost_tree()).

The simplest form would probably just be a collection of tables (one per model), with columns for each supported engine + a final column corresponding to the parsnip parameter names.

Since this could take a bit of time to add, perhaps if you think this sort of thing would be useful, you could decide on an appropriate location/format, and add placeholders.

Then, myself and other users could help fill it in as we use the models.

**Comment by khughitt** [comment]
Actually, it looks like the necessary mappings could probably be parsed from the R code (e.g. `boost_tree_data.R`) in a pretty straight-forward manner. Might be quicker / less error-prone than attempting to manually generate the tables..

**Comment by thohan88** [comment]
I just opened a similar issue (#238) for the exact same purpose. Based on #235, I learned you can use `parsnip::get_model_env` to list all arguments. I think it would be useful if the mapping between original and standardized arguments was documented in the function itself and/or in a complete reference table or cheat sheet as suggested, e.g. for boosted trees:

|model      |parsnip        |C5.0     |spark                   |xgboost          |
  |:----------|:--------------|:--------|:-----------------------|:----------------|
  |boost_tree |learn_rate     |NA       |step_size               |eta              |
  |boost_tree |loss_reduction |NA       |NA                      |gamma            |
  |boost_tree |min_info_gain  |NA       |gamma                   |NA               |
  |boost_tree |min_n          |minCases |min_instances_per_node  |min_child_weight |
  |boost_tree |mtry           |NA       |feature_subset_strategy |colsample_bytree |
  |boost_tree |sample_size    |sample   |subsampling_rate        |subsample        |
  |boost_tree |tree_depth     |NA       |max_depth               |max_depth        |
  |boost_tree |trees          |trials   |max_iter                |nrounds          |

Example code for generating said tables below.

<details>
  <summary>Code</summary>
  
  ```
library(tidyverse)
envir <- parsnip::get_model_env()

args <- ls(envir) %>% 
  tibble(name = .) %>% 
  filter(str_detect(name, "args")) %>% 
  mutate(model = str_replace(name, "_args", ""),
         args  = map(name, ~envir[[.x]])) %>% 
  unnest(args) %>% 
  select(model:original)
```
</details>
  
<details>
<summary>Full mapping table</summary>

```
params %>% 
  knitr::kable()
```

|model            |engine       |parsnip         |original                |
  |:----------------|:------------|:---------------|:-----------------------|
  |boost_tree       |xgboost      |tree_depth      |max_depth               |
  |boost_tree       |xgboost      |trees           |nrounds                 |
  |boost_tree       |xgboost      |learn_rate      |eta                     |
  |boost_tree       |xgboost      |mtry            |colsample_bytree        |
  |boost_tree       |xgboost      |min_n           |min_child_weight        |
  |boost_tree       |xgboost      |loss_reduction  |gamma                   |
  |boost_tree       |xgboost      |sample_size     |subsample               |
  |boost_tree       |C5.0         |trees           |trials                  |
  |boost_tree       |C5.0         |min_n           |minCases                |
  |boost_tree       |C5.0         |sample_size     |sample                  |
  |boost_tree       |spark        |tree_depth      |max_depth               |
  |boost_tree       |spark        |trees           |max_iter                |
  |boost_tree       |spark        |learn_rate      |step_size               |
  |boost_tree       |spark        |mtry            |feature_subset_strategy |
  |boost_tree       |spark        |min_n           |min_instances_per_node  |
  |boost_tree       |spark        |min_info_gain   |gamma                   |
  |boost_tree       |spark        |sample_size     |subsampling_rate        |
  |decision_tree    |rpart        |tree_depth      |maxdepth                |
  |decision_tree    |rpart        |min_n           |minsplit                |
  |decision_tree    |rpart        |cost_complexity |cp                      |
  |decision_tree    |C5.0         |min_n           |minCases                |
  |decision_tree    |spark        |tree_depth      |max_depth               |
  |decision_tree    |spark        |min_n           |min_instances_per_node  |
  |linear_reg       |glmnet       |penalty         |lambda                  |
  |linear_reg       |glmnet       |mixture         |alpha                   |
  |linear_reg       |spark        |penalty         |reg_param               |
  |linear_reg       |spark        |mixture         |elastic_net_param       |
  |logistic_reg     |glmnet       |penalty         |lambda                  |
  |logistic_reg     |glmnet       |mixture         |alpha                   |
  |logistic_reg     |spark        |penalty         |reg_param               |
  |logistic_reg     |spark        |mixture         |elastic_net_param       |
  |logistic_reg     |keras        |decay           |decay                   |
  |mars             |earth        |num_terms       |nprune                  |
  |mars             |earth        |prod_degree     |degree                  |
  |mars             |earth        |prune_method    |pmethod                 |
  |mlp              |keras        |hidden_units    |hidden_units            |
  |mlp              |keras        |penalty         |penalty                 |
  |mlp              |keras        |dropout         |dropout                 |
  |mlp              |keras        |epochs          |epochs                  |
  |mlp              |keras        |activation      |activation              |
  |mlp              |nnet         |hidden_units    |size                    |
  |mlp              |nnet         |penalty         |decay                   |
  |mlp              |nnet         |epochs          |maxit                   |
  |multinom_reg     |glmnet       |penalty         |lambda                  |
  |multinom_reg     |glmnet       |mixture         |alpha                   |
  |multinom_reg     |spark        |penalty         |reg_param               |
  |multinom_reg     |spark        |mixture         |elastic_net_param       |
  |multinom_reg     |keras        |decay           |decay                   |
  |nearest_neighbor |kknn         |neighbors       |ks                      |
  |nearest_neighbor |kknn         |weight_func     |kernel                  |
  |nearest_neighbor |kknn         |dist_power      |distance                |
  |rand_forest      |ranger       |mtry            |mtry                    |
  |rand_forest      |ranger       |trees           |num.trees               |
  |rand_forest      |ranger       |min_n           |min.node.size           |
  |rand_forest      |randomForest |mtry            |mtry                    |
  |rand_forest      |randomForest |trees           |ntree                   |
  |rand_forest      |randomForest |min_n           |nodesize                |
  |rand_forest      |spark        |mtry            |feature_subset_strategy |
  |rand_forest      |spark        |trees           |num_trees               |
  |rand_forest      |spark        |min_n           |min_instances_per_node  |
  |surv_reg         |flexsurv     |dist            |dist                    |
  |surv_reg         |survival     |dist            |dist                    |
  |svm_poly         |kernlab      |cost            |C                       |
  |svm_poly         |kernlab      |degree          |degree                  |
  |svm_poly         |kernlab      |scale_factor    |scale                   |
  |svm_poly         |kernlab      |margin          |epsilon                 |
  |svm_rbf          |kernlab      |cost            |C                       |
  |svm_rbf          |kernlab      |rbf_sigma       |sigma                   |
  |svm_rbf          |kernlab      |margin          |epsilon                 |
  </details>
  
  <details>
  <summary>Mapping table by model</summary>
  
  ```
args %>% 
  split(.$model) %>% 
  map(spread, engine, original) %>% 
  map(knitr::kable)
```

$boost_tree

|model      |parsnip        |C5.0     |spark                   |xgboost          |
  |:----------|:--------------|:--------|:-----------------------|:----------------|
  |boost_tree |learn_rate     |NA       |step_size               |eta              |
  |boost_tree |loss_reduction |NA       |NA                      |gamma            |
  |boost_tree |min_info_gain  |NA       |gamma                   |NA               |
  |boost_tree |min_n          |minCases |min_instances_per_node  |min_child_weight |
  |boost_tree |mtry           |NA       |feature_subset_strategy |colsample_bytree |
  |boost_tree |sample_size    |sample   |subsampling_rate        |subsample        |
  |boost_tree |tree_depth     |NA       |max_depth               |max_depth        |
  |boost_tree |trees          |trials   |max_iter                |nrounds          |
  
  $decision_tree

|model         |parsnip         |C5.0     |rpart    |spark                  |
  |:-------------|:---------------|:--------|:--------|:----------------------|
  |decision_tree |cost_complexity |NA       |cp       |NA                     |
  |decision_tree |min_n           |minCases |minsplit |min_instances_per_node |
  |decision_tree |tree_depth      |NA       |maxdepth |max_depth              |
  
  $linear_reg


|model      |parsnip |glmnet |spark             |
  |:----------|:-------|:------|:-----------------|
  |linear_reg |mixture |alpha  |elastic_net_param |
  |linear_reg |penalty |lambda |reg_param         |
  
  $logistic_reg


|model        |parsnip |glmnet |keras |spark             |
  |:------------|:-------|:------|:-----|:-----------------|
  |logistic_reg |decay   |NA     |decay |NA                |
  |logistic_reg |mixture |alpha  |NA    |elastic_net_param |
  |logistic_reg |penalty |lambda |NA    |reg_param         |
  
  $mars


|model |parsnip      |earth   |
  |:-----|:------------|:-------|
  |mars  |num_terms    |nprune  |
  |mars  |prod_degree  |degree  |
  |mars  |prune_method |pmethod |
  
  $mlp


|model |parsnip      |keras        |nnet  |
  |:-----|:------------|:------------|:-----|
  |mlp   |activation   |activation   |NA    |
  |mlp   |dropout      |dropout      |NA    |
  |mlp   |epochs       |epochs       |maxit |
  |mlp   |hidden_units |hidden_units |size  |
  |mlp   |penalty      |penalty      |decay |
  
  $multinom_reg


|model        |parsnip |glmnet |keras |spark             |
  |:------------|:-------|:------|:-----|:-----------------|
  |multinom_reg |decay   |NA     |decay |NA                |
  |multinom_reg |mixture |alpha  |NA    |elastic_net_param |
  |multinom_reg |penalty |lambda |NA    |reg_param         |
  
  $nearest_neighbor


|model            |parsnip     |kknn     |
  |:----------------|:-----------|:--------|
  |nearest_neighbor |dist_power  |distance |
  |nearest_neighbor |neighbors   |ks       |
  |nearest_neighbor |weight_func |kernel   |
  
  $rand_forest


|model       |parsnip |randomForest |ranger        |spark                   |
  |:-----------|:-------|:------------|:-------------|:-----------------------|
  |rand_forest |min_n   |nodesize     |min.node.size |min_instances_per_node  |
  |rand_forest |mtry    |mtry         |mtry          |feature_subset_strategy |
  |rand_forest |trees   |ntree        |num.trees     |num_trees               |
  
  $surv_reg


|model    |parsnip |flexsurv |survival |
  |:--------|:-------|:--------|:--------|
  |surv_reg |dist    |dist     |dist     |
  
  $svm_poly


|model    |parsnip      |kernlab |
  |:--------|:------------|:-------|
  |svm_poly |cost         |C       |
  |svm_poly |degree       |degree  |
  |svm_poly |margin       |epsilon |
  |svm_poly |scale_factor |scale   |
  
  $svm_rbf


|model   |parsnip   |kernlab |
  |:-------|:---------|:-------|
  |svm_rbf |cost      |C       |
  |svm_rbf |margin    |epsilon |
  |svm_rbf |rbf_sigma |sigma   |

**Comment by topepo** [comment]
Thanks! I added a subsection to the new [`Models` vignette](https://tidymodels.github.io/parsnip/dev/articles/articles/Models.html) using your code. Let me know if you want any changes.

**Comment by juliasilge** [comment]
Just noticed nothing is showing up for stan in the existing tables.

**Comment by topepo** [comment]
No tuning parameters for those models. The same happens in `linear_reg()` since there is no `lm` column.
